### PR TITLE
feat: update crosschain gov v0.47

### PR DESCRIPTION
**Pull request by alexgao001**
### Description

This pr adds cross-chain gov for 0.47 upgrade. Contains following previous prs:

+ https://github.com/bnb-chain/greenfield-cosmos-sdk/pull/110
+ https://github.com/bnb-chain/greenfield-cosmos-sdk/pull/142

### Rationale

for cross-chain gov 

### Example

NA

### Changes

Notable changes:
* cross-chain gov

### CODE REVIEW

File: x/gov/types/v1/msgs.go [hunk 1]
```diff
@@ -307,6 +307,26 @@ func (msg MsgUpdateParams) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{authority}
 }
 
+// ValidateBasic implements the sdk.Msg interface.
+func (msg MsgUpdateCrossChainParams) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
```
Comment (yutianwu) [inline x/gov/types/v1/msgs.go:312]: don't use Bech32 now

File: x/gov/types/crosschain.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	SyncParamsChannel                 = "syncParametersChange"
+	SyncParamsChannelID sdk.ChannelID = 3
+	KeyUpgrade                        = "upgrade"
+)
+
+var KeySyncParamsRelayerFee = []byte("SyncParamsRelayerFee")
```
Comment (yutianwu) [inline x/gov/types/crosschain.go:13]: delete unused key

File: x/gov/types/v1/msgs.go (hunk 1 above)
Comment (alexgao001) [reply x/gov/types/v1/msgs.go:312]: fixed

File: x/gov/types/crosschain.go (hunk 2 above)
Comment (alexgao001) [reply x/gov/types/crosschain.go:13]: deleted

**Review by yutianwu — APPROVED** [review]

File: x/gov/keeper/crosschain.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		sdk.SynCrossChainPackageType,
+		encodedPackage,
+		big.NewInt(0),
+		big.NewInt(0),
+	)
+	return err
+}
+
+func (k Keeper) ExecuteSynPackage(ctx sdk.Context, _ *sdk.CrossChainAppContext, payload []byte) sdk.ExecuteResult {
+	k.Logger(ctx).Error("received sync params sync package", "payload", hex.EncodeToString(payload))
```
Comment (forcodedancing) [inline x/gov/keeper/crosschain.go:63]: log with `Error`? Seems the log is for `Info` or `Debug` here?

File: x/gov/keeper/crosschain.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	return sdk.ExecuteResult{}
+}
+
+func (k Keeper) ExecuteAckPackage(ctx sdk.Context, _ *sdk.CrossChainAppContext, payload []byte) sdk.ExecuteResult {
+	k.Logger(ctx).Error("received sync params in ack package", "payload", hex.EncodeToString(payload))
```
Comment (forcodedancing) [inline x/gov/keeper/crosschain.go:68]: log with `Error`? Seems the log is for `Info` or `Debug` here?

File: x/gov/keeper/crosschain.go [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	return sdk.ExecuteResult{}
+}
+
+func (k Keeper) ExecuteFailAckPackage(ctx sdk.Context, _ *sdk.CrossChainAppContext, payload []byte) sdk.ExecuteResult {
+	k.Logger(ctx).Error("received sync params fail ack package", "payload", hex.EncodeToString(payload))
```
Comment (forcodedancing) [inline x/gov/keeper/crosschain.go:73]: log with `Error`? Seems the log is for `Info` or `Debug` here?

File: x/gov/keeper/crosschain.go (hunk 3 above)
Comment (forcodedancing) [reply x/gov/keeper/crosschain.go:63]: Why there is no logic here? `return sdk.ExecuteResult{}`

File: x/gov/keeper/msg_server.go [hunk 6]
```diff
@@ -83,7 +83,7 @@ func (k msgServer) SubmitProposal(goCtx context.Context, msg *v1.MsgSubmitPropos
 	}, nil
 }
 
-// CancelProposals implements the MsgServer.CancelProposal method.
+// CancelProposal CancelProposals implements the MsgServer.CancelProposal method.
```
Comment (forcodedancing) [inline x/gov/keeper/msg_server.go:86]: `CancelProposal implements the MsgServer.CancelProposal method.`

Comment (alexgao001) [reply x/gov/keeper/msg_server.go]: fixed

**Review by forcodedancing — APPROVED** [review]

**Pull request merged**